SwiftCompilerSources/Sources/SIL/ForwardingInstruction.swift

```diff
@@ -0,0 +1,155 @@
+//===--- ForwardingInstruction.swift - forwarding instruction protocols ---===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SILBridging
+
+/// An instruction that forwards ownership from operands to results.
+public protocol ForwardingInstruction : Instruction {}
+
+extension ForwardingInstruction {
+  public var singleForwardedOperand: Operand? {
+    switch self {
```

Comment:
As `ForwardingInstruction` is a protocol, it's better to not hand code the protocol dispatch, but add `singleForwardedOperand` as a protocol requirement and add conforming methods for all these instruction types (can be done in this file with extensions).